src/effects/deeplearning/DeepModel.h

```diff
@@ -0,0 +1,82 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+   License: GPL v2.  See License.txt.
+
+   DeepModel.h
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+\class DeepModel
+\brief base class for handling torchscript models
+
+*/
+/*******************************************************************/
+
+#pragma once
+
+#include "ModelCard.h"
+
+#include <torch/script.h>
+#include <torch/torch.h>
+
+using ModulePtr = std::unique_ptr<torch::jit::script::Module>;
+
+class ModelException : public std::exception
```

Comment:
Probably MessageBoxException is a better choice for the base here. Looks like Resample, for example, is called outside try{}catch{} (Or I missed something)